src/daemon/rpmostree-sysroot-core.cxx

```diff
@@ -376,11 +377,8 @@ rpmostree_syscore_get_origin_merge_deployment (OstreeSysroot *self, const char *
 gboolean 
 rpmostree_syscore_livefs_query (OstreeSysroot *self, OstreeDeployment *deployment, gboolean *out_is_live, GError **error)
 {
-  g_autofree char *live_inprogress = NULL;
-  g_autofree char *live_replaced = NULL;
-  if (!ror_livefs_get_state (self, deployment, &live_inprogress, &live_replaced, error))
-    return FALSE;
-  *out_is_live = (live_inprogress != NULL) || (live_replaced != NULL);
+  auto live_state = rpmostreecxx::get_live_apply_state(*self, *deployment);
+  *out_is_live = (live_state.inprogress.length() > 0) || (live_state.commit.length() > 0);
```

Comment:
Minor: I think this may benefit from an `live_state.is_live()` method.

Reply:
I don't think it's possible (yet?) to natively define bridged methods on "shared" types (i.e. not `extern`).  On the Rust side we can define traits, but that isn't relevant here because this shared struct is already mostly a workaround for the lack of `Option<>` bindings.

I guess we could make it an opaque Rust type with accessors?  

Or hmm...might be possible C++ side to declare a method external to the struct.  But on the other hand...in practice we still on the C++ side need to be aware of the "internals" of this data (mainly for the `status` part and conversion to GVariant) so we wouldn't be encapsulating really.